### ISSUE TITLE
sqrtf and sqrt in device code cannot be compiled

### ISSUE THREAD

**Issue by ex-rzr** [GPUOpen-ProfessionalCompute-Tools/HIP#70]
To reproduce replace line 46 (https://github.com/GPUOpen-ProfessionalCompute-Tools/HIP/blob/master/samples/0_Intro/square/square.hipref.cpp#L46)
with
```c
        C_d[i] = sqrtf(A_d[i]);
```

`hipcc  square.hipref.cpp -o square.hip.out` fails with
```
Referencing function in another module!
  %34 = call float @sqrtf(float %33) #5
LLVM ERROR: Broken function found, compilation aborted!
Generating HSAIL BRIG kernel failed
```

Full log: [hipcc-sqrt-log.txt](https://github.com/GPUOpen-ProfessionalCompute-Tools/HIP/files/961435/hipcc-sqrt-log.txt)

I don't know about other functions but the same error for `sqrt(double)`.
But, for example, this can be compiled:
```c
        C_d[i] = __fsqrt_rn(A_d[i]);
```

Checked on:
```
HIP version: 1.0.16503
HCC clang version 3.5.0  (based on HCC 0.10.16501-81f0a2f-02246a0 LLVM 3.5.0svn)
```

**Comment by adityaatluri** [comment]
Hi,
You are using old version of HCC and HIP which have bugs related to math functions. We fixed it with https://github.com/GPUOpen-ProfessionalCompute-Tools/HIP/releases/tag/preview_1.0.17102

**Comment by ex-rzr** [comment]
Thank you, I'll check the new version tomorrow.

**Comment by adityaatluri** [comment]
Hi,
We released ROCm 1.5 (https://github.com/RadeonOpenCompute/ROCm). I recommend using it.

**Comment by ex-rzr** [comment]
Sorry for delay. It works on ROCm 1.5.
I'll close the issue. Thanks!
